src/shell/shell_cmds.cpp

```diff
@@ -2242,3 +2243,290 @@ void DOS_Shell::CMD_VOL(char* args)
 	const auto trimmed_label = trim(label);
 	WriteOut(MSG_Get("SHELL_CMD_VOL_OUTPUT"), drive, trimmed_label, high, low);
 }
+
+static std::vector<std::string> cmd_move_parse_sources(const char*& args)
+{
+	bool parsing             = true;
+	bool inside_quote        = false;
+	bool stop_next_letter    = false;
+	bool continue_after_stop = false;
+	std::string word;
+	std::vector<std::string> sources;
+
+	// Sources are seperated by commas and terminated by a trailing space
+	while (parsing) {
+		switch (*args) {
+		case '"':
+			inside_quote = !inside_quote;
+			args++;
+			break;
+		case ' ':
+			if (inside_quote) {
+				word.push_back(*args++);
+			} else {
+				if (!word.empty()) {
+					sources.emplace_back(std::move(word));
+					word             = {};
+					stop_next_letter = true;
+					continue_after_stop = false;
+				}
+				args++;
+			}
+			break;
+		case ',':
+			stop_next_letter = true;
+			continue_after_stop = true;
+			if (!word.empty()) {
+				sources.emplace_back(std::move(word));
+				word = {};
+			}
+			args++;
+			break;
+		case 0:
+			if (!word.empty()) {
+				sources.emplace_back(std::move(word));
+			}
+			parsing = false;
+			break;
+		default:
+			if (stop_next_letter) {
+				parsing             = continue_after_stop;
+				continue_after_stop = false;
+				stop_next_letter    = false;
+			} else {
+				word.push_back(*args++);
+			}
+		}
+	}
+	return sources;
+}
+
+static std::string cmd_move_parse_destination(const char*& args)
+{
+	std::string destination;
+	bool parsing      = true;
+	bool inside_quote = false;
+
+	while (parsing) {
+		switch (*args) {
+		case '"':
+			inside_quote = !inside_quote;
+			args++;
+			break;
+		case ' ':
+			if (inside_quote) {
+				destination.push_back(*args++);
+			} else {
+				args++;
+				parsing = false;
+			}
+			break;
+		case '\0': parsing = false; break;
+		default: destination.push_back(*args++);
+		}
+	}
+	return destination;
+}
+
+struct cmd_move_arguments {
+	std::vector<std::string> sources = {};
+	std::string destination          = {};
+	const char* error                = nullptr;
+};
```

Comment:
🎖️ for using a struct for multiple return types (I see it as the "poor man's data class" in C++ 😅 )